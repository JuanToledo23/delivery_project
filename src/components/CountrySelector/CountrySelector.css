.radio-container {
  width: 176px;
  height: 100px;
  box-shadow: 0px 0px 2px rgba(0, 0, 0, 0.14), 0px 2px 2px rgba(0, 0, 0, 0.12), 0px 1px 3px rgba(0, 0, 0, 0.2);
  border-radius: 8px;
  background: #FFFFFF;
  padding: 12px 16px;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-right: 16px;
  cursor: pointer;
  margin-top: 16px;
}

.radio-container > .flag-container {
  display: flex;
  flex-direction: column;
}

.radio-container > .flag-container > img {
  width: 43.51px;
}

.radio-container > .flag-container > span {
  margin-top: 8px;
}

.countries-container {
  display: flex;
}

.radio-container:hover {
  background-color: rgb(231 255 247);
}
