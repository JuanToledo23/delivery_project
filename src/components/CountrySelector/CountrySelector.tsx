import { useEffect, useState } from "react";
import { Radio } from "@mui/material";
import "./CountrySelector.css";
import { ApiCountry } from "models/Api";

type CountrySelectorProps = {
  onSelect?: Function;
  name?: string;
  countries?: Array<ApiCountry>;
};

function CountrySelector({ onSelect, name, countries }: CountrySelectorProps) {
  const [selectedValue, setSelectedValue] = useState(1);

  const handleChange = (id: any) => {
    setSelectedValue(id);
  };

  useEffect(() => {
    if (onSelect && selectedValue) {
      onSelect({ selectedValue, name });
    }
  }, [selectedValue]);

  return (
    <div className="countries-container" data-testid="CountrySelector">
      {countries && countries?.length > 0 ? (
        <>
          {countries?.map((country) => {
            return (
              <div
                className="radio-container"
                key={country.id}
                onClick={() => handleChange(country.id)}
                id={`rad_${country.id}`}
              >
                <div className="flag-container">
                  <img src={`/flags/${country.flag}`} alt={country.name} />
                  <span className="body1" data-testid="list-countries">
                    {country.name}
                  </span>
                </div>
                <Radio
                  checked={selectedValue === country.id}
                  value={country.id}
                  name={name}
                  inputProps={{ "aria-label": country.name }}
                  className="radio"
                  data-testid="radio-test"
                />
              </div>
            );
          })}
        </>
      ) : (
        <>
          <div className="skeleton box-shadow" style={{ maxWidth: "208px" }}>
            <div className="skeleton-left">
              <div className="circle"></div>
              <div className="line w80" style={{ marginBottom: "10px" }}></div>
            </div>
          </div>
          <div className="skeleton box-shadow" style={{ maxWidth: "208px" }}>
            <div className="skeleton-left">
              <div className="circle"></div>
              <div className="line w80" style={{ marginBottom: "10px" }}></div>
            </div>
          </div>
          <div className="skeleton box-shadow" style={{ maxWidth: "208px" }}>
            <div className="skeleton-left">
              <div className="circle"></div>
              <div className="line w80" style={{ marginBottom: "10px" }}></div>
            </div>
          </div>
        </>
      )}
    </div>
  );
}

export default CountrySelector;
