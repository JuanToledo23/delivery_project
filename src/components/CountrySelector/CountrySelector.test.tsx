import { render, screen } from '@testing-library/react';
import CountrySelector from './CountrySelector';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

const countries = [
  {
    id: 1,
    name: 'México',
    flag: "mexico.svg"
  },
  {
    id: 2,
    name: 'Colombia',
    flag: "colombia.svg"
  },
  {
    id: 3,
    name: 'USA',
    flag: "usa.svg"
  }
];

describe('CountrySelector', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <CountrySelector />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("render countries correctly", async () => {
    const listCountries = screen.getAllByTestId("list-countries");
    expect(listCountries[0]).toHaveTextContent("México");
  });

  it("onclick event radios", async () => {
    const handleChange = jest.fn();
    render(
      <CountrySelector
        countries={countries}
        onSelect={handleChange}
      />
    );
    const radioButton = screen.getAllByTestId("radio-test");
    userEvent.click(radioButton[0]);
    expect(handleChange).toHaveBeenCalled();
  });
});