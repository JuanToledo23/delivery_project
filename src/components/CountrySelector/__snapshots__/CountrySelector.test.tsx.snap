// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelector should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="countries-container"
    data-testid="CountrySelector"
  >
    <div
      class="radio-container"
      id="rad_1"
    >
      <div
        class="flag-container"
      >
        <img
          alt="México"
          src="/flags/mexico.svg"
        />
        <span
          class="body1"
          data-testid="list-countries"
        >
          México
        </span>
      </div>
      <span
        class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary radio css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        data-testid="radio-test"
      >
        <input
          aria-label="México"
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          type="radio"
          value="1"
        />
        <span
          class="css-hyxlzm"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
    </div>
    <div
      class="radio-container"
      id="rad_2"
    >
      <div
        class="flag-container"
      >
        <img
          alt="Colombia"
          src="/flags/colombia.svg"
        />
        <span
          class="body1"
          data-testid="list-countries"
        >
          Colombia
        </span>
      </div>
      <span
        class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary radio css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        data-testid="radio-test"
      >
        <input
          aria-label="Colombia"
          class="PrivateSwitchBase-input css-1m9pwf3"
          type="radio"
          value="2"
        />
        <span
          class="css-hyxlzm"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
    </div>
    <div
      class="radio-container"
      id="rad_3"
    >
      <div
        class="flag-container"
      >
        <img
          alt="USA"
          src="/flags/usa.svg"
        />
        <span
          class="body1"
          data-testid="list-countries"
        >
          USA
        </span>
      </div>
      <span
        class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary radio css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        data-testid="radio-test"
      >
        <input
          aria-label="USA"
          class="PrivateSwitchBase-input css-1m9pwf3"
          type="radio"
          value="3"
        />
        <span
          class="css-hyxlzm"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
